### PR TITLE
[buddy] Add an optional PodMonitor to the teleport-kube-agent chart

### LINKED ISSUE #30944: Add podMonitor to teleport-kube-agent chart
# What would you like Teleport to do?
Add in a podmonitor to the teleport-kube-agent chart (as exists in teleport-cluster chart). 
I already have a PR prepped

# What problem does this solve?
Allows for monitoring of teleport using prometheus-operator

### PR DESCRIPTION
**Pull request by hugoShaka**
Fixes https://github.com/gravitational/teleport/issues/30944

Buddy PR for #30945

This PR also adds a documentation commit adding the field to the chart reference (it was also missing from `teleport-cluster`).

### CODE REVIEW

File: docs/pages/reference/helm-reference/teleport-cluster.mdx [hunk 1]
```diff
@@ -804,6 +804,46 @@ recommended to use one of the other modes and rely on
 to inject your custom configuration.
 </Admonition>
 
+## `podMonitor`
+
+`podMonitor` controls [the PodMonitor CR (from monitoring.coreos.com/v1)
+](https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/api.md#monitoring.coreos.com/v1.PodMonitor)
+This custom resource is used to configure Prometheus to monitor workload running
```
Comment (strideynet) [inline docs/pages/reference/helm-reference/teleport-cluster.mdx:811]: I think we can improve this help text - at no point does it really say that it'll enable monitoring of the workloads deployed by this chart 😅

**Review by strideynet — APPROVED** [review]
Looks good but docs could be clearer

**Review by zmb3 — APPROVED** [review]

**Review by tigrato — APPROVED** [review]

File: docs/pages/reference/helm-reference/teleport-kube-agent.mdx [hunk 2]
```diff
@@ -1080,6 +1080,46 @@ Ensures that this number of replicas is available during voluntary disruptions,
       minAvailable: 1
   ```
 
+## `podMonitor`
+
+`podMonitor` controls [the PodMonitor CR (from monitoring.coreos.com/v1)
+](https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/api.md#monitoring.coreos.com/v1.PodMonitor)
+This custom resource is used to configure Prometheus to monitor workload running
+in Kubernetes. This value can be used if you are using both Prometheus and
+prometheus-operator to monitor workload running in the Kubernetes cluster.
```
Comment (hugoShaka) [inline docs/pages/reference/helm-reference/teleport-kube-agent.mdx:1089]:
```suggestion
that monitors the workload (auth and proxy pods) deployed by the chart.
This custom resource configures Prometheus and makes it scrape Teleport metrics.
```

File: docs/pages/reference/helm-reference/teleport-cluster.mdx [hunk 3]
```diff
@@ -804,6 +804,46 @@ recommended to use one of the other modes and rely on
...
+in Kubernetes. This value can be used if you are using both Prometheus and
+prometheus-operator to monitor workload running in the Kubernetes cluster.
```
Comment (hugoShaka) [inline docs/pages/reference/helm-reference/teleport-cluster.mdx:813]:
```suggestion
that monitors the workload (auth and proxy pods) deployed by the chart.
This custom resource configures Prometheus and makes it scrape Teleport metrics.
```

File: docs/pages/reference/helm-reference/teleport-cluster.mdx [hunk 4]
```diff
@@ -804,6 +804,45 @@ recommended to use one of the other modes and rely on
 to inject your custom configuration.
 </Admonition>
 
+## `podMonitor`
+
+`podMonitor` controls [the PodMonitor CR (from monitoring.coreos.com/v1)
+](https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/api.md#monitoring.coreos.com/v1.PodMonitor)
+that monitors the workload (auth and proxy pods) deployed by the chart.
```
Comment (hugoShaka) [inline docs/pages/reference/helm-reference/teleport-cluster.mdx:811]:
```suggestion
that monitors the workload (Auth and Proxy Services) deployed by the chart.
```

File: docs/pages/reference/helm-reference/teleport-kube-agent.mdx [hunk 5]
```diff
@@ -1080,6 +1080,45 @@ Ensures that this number of replicas is available during voluntary disruptions,
       minAvailable: 1
   ```
 
+## `podMonitor`
+
+`podMonitor` controls [the PodMonitor CR (from monitoring.coreos.com/v1)
+](https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/api.md#monitoring.coreos.com/v1.PodMonitor)
+that monitors the workload (auth and proxy pods) deployed by the chart.
```
Comment (hugoShaka) [inline docs/pages/reference/helm-reference/teleport-kube-agent.mdx:1087]:
```suggestion
that monitors the workload (Auth and Proxy Services) deployed by the chart.
```

**Pull request merged**